src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowCopyAction.groovy

```diff
@@ -530,6 +532,11 @@ class ShadowCopyAction implements CopyAction {
             return archivePath.entry.unixMode
         }
 
+        @Override
+        FilePermissions getPermissions() {
```

Comment:
The `getMode()` method above, which this one is supposed to replace still has usages in this same class, higher up. We should replace those usages.

Reply:
Is this cd06568de744cfb368239cef1f6e3a2d64daa001 right?

Reply:
I'm sorry. Now I realize I was completely wrong. I think the correct version is:

```groovy
@Override
int getMode() {
    return archivePath.entry.unixMode
}

@Override
FilePermissions getPermissions() {
    return new DefaultFilePermissions(getMode())
}
```

So `getMode()` remains unchanged, `getPermissions()` uses it. Does it make sense to you too?
Sorry for the confusion, I'm used to looking at it the other way around...

Reply:
That sounds resonable. Thanks for your explanation!